### PR TITLE
Feature/abx 4313

### PR DESCRIPTION
**Pull request by Sam-Jeston**

### CODE REVIEW

File: src/components/Dashboard.tsx [hunk 1]
```diff
@@ -21,5 +25,35 @@ export const Dashboard: React.SFC = () => (
       </div>
     </div>
     <Transactions />
-  </div>
+  </React.Fragment>
```
Comment (Carl-Foster) [inline src/components/Dashboard.tsx]: I think this is what broke your styling

File: src/components/Dashboard.tsx [hunk 2]
```diff
@@ -21,5 +25,35 @@ export const Dashboard: React.SFC = () => (
...
+      case AccountPageEnum.transfer: return <TransferPage />
+      case AccountPageEnum.sign: return <Sign />
+      default: return <div />
+    }
+  }
+
+  render() {
+    return (
+      <div className='vertical-spaced'>
+        <div className='columns'>
```
Comment (Carl-Foster) [inline src/components/Dashboard.tsx]: I reckon you should make this a `level` component

File: src/components/Sign.tsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+            helpText='Enter the text to sign'
+            onChangeHandler={(newValue) => handleSignFormChange('message', newValue)}
+          />
+          <div className='field is-grouped'>
+            <div className='control is-expanded'>
+              <button className='button is-fullwidth' type='submit'>Sign</button>
+            </div>
+          </div>
+        </form>
+        <div className='field' style={{marginTop: '20px'}}>
```
Comment (Carl-Foster) [inline src/components/Sign.tsx]: You could just add the field into the form and not require the `marginTop`

File: src/helpers/enumStringValues.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export function enumStringValues(enumerable: any) {
```
Comment (Carl-Foster) [inline src/helpers/enumStringValues.ts]: Could this be 
```ts
function enumStringValues(enumerable: Record<string, string | number>): string[] {
    return Object.entries(enumerable).map(([, val]) => val).filter((val) => typeof val === 'string')
}
```

File: src/store/reducers/sign.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      default: return state
+    }
+  },
+})
+
+function handleChange(name: keyof SignedMessage) {
+  return (state = '', action: RootAction) => {
+    switch (action.type) {
+      case getType(updateSignForm):
+        return action.payload.field === name ? action.payload.newValue : state
```
Comment (Carl-Foster) [inline src/store/reducers/sign.ts]: could get rid of this line :+1:

File: src/components/Dashboard.tsx [hunk 6]
```diff
@@ -21,5 +25,35 @@ export const Dashboard: React.SFC = () => (
       </div>
     </div>
     <Transactions />
-  </div>
+  </React.Fragment>
 )
+
+export class Dashboard extends React.PureComponent<Props> {
```
Comment (Carl-Foster) [inline src/components/Dashboard.tsx]: I'd change this and call it `AccountPage` now.

File: src/components/Dashboard.tsx [hunk 7]
```diff
@@ -1,17 +1,21 @@
...
-export const Dashboard: React.SFC = () => (
-  <div className='vertical-spaced'>
-    <div className='has-text-centered title-heading is-constant-height'>
-      <Password />
-    </div>
+export interface Props {
+  accountPage: AccountPageEnum
+}
+
+export const TransferPage: React.SFC = () => (
```
Comment (Carl-Foster) [inline src/components/Dashboard.tsx]: I'd also make this `Dashboard` or `AccountDashboard`

Comment (Carl-Foster) [reply src/components/Dashboard.tsx]: And split it into a separate file.

File: src/components/Dashboard.tsx (hunk 1 above)
Comment (Sam-Jeston) [reply src/components/Dashboard.tsx]: yeh it's not this...

File: src/helpers/enumStringValues.ts (hunk 4 above)
Comment (Sam-Jeston) [reply src/helpers/enumStringValues.ts]: blah

File: src/components/Sign.tsx [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+  }
+
+  validateProps = (): void | never => {
+    this.checkWalletIsUnlocked()
+    this.checkValidEntry('message')
+  }
+
+  checkValidEntry = (key: keyof SignedMessage) => {
+    if (!this.props.signData[key]) {
+      throw new InputError(`${startCase(key)} is required`, `payee-${kebabCase(key)}`)
```
[the remark this answers is not in the document]
Comment (Sam-Jeston) [reply src/components/Sign.tsx]: 🚀

[the remark this answers is not in the document]
Comment (Sam-Jeston) [reply src/components/Sign.tsx]: Also. This validates my point of generic forms for this kind of crud

File: src/components/Dashboard.tsx (hunk 1 above)
Comment (Sam-Jeston) [reply src/components/Dashboard.tsx]: Sorted it

File: src/components/Dashboard.tsx (hunk 2 above)
Comment (Sam-Jeston) [reply src/components/Dashboard.tsx]: Looks mint

File: src/helpers/enumStringValues.ts (hunk 4 above)
Comment (Sam-Jeston) [reply src/helpers/enumStringValues.ts]: I copied this off stackoverflow. How dare you change its sacred nature.

File: src/store/reducers/sign.ts (hunk 5 above)
Comment (Sam-Jeston) [reply src/store/reducers/sign.ts]: indeed sir!

File: src/components/Dashboard.tsx (hunk 6 above)
Comment (Sam-Jeston) [reply src/components/Dashboard.tsx]: done

File: src/components/Dashboard.tsx (hunk 7 above)
Comment (Sam-Jeston) [reply src/components/Dashboard.tsx]: yep done

**Pull request merged**